// pages/gameOver/components/resultsTable.js
import React from 'react'
import 'css/global.css'

const ResultsTable = () => {

    let data = [
        {
            id: 0,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 1,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 2,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 3,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: false
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }, {
            id: 4,
            name: 'utoomey',
            score: 100,
            lines: 100,
            level: 100,
            isPlay: true
        }
    ]

    data = data.sort((a, b) => {
        if (!(a.isPlay) && b.isPlay) {
            return 0
        }
        return -1
    })

    const isAllCameOut = !(data.some(obj => obj.isPlay))

    console.log(data)
    return (
        <div className='box box_size-w_big'>
            <div className='table'>
                <div className='line'>
                    <p className='text location_horizontal-center text_size_big'>Game Over!</p>
                </div>
                <hr className='box__hr-h'></hr>
                <div className='location_horizontal'>
                    <p className='text table__text_size-w_index text_size_small'>№</p>
                    <div className='table__hr'/>
                    <p className='text table__text_size-w_name text_size_small'>name</p>
                    <div className='table__hr'/>
                    <p className='text table__text_size-w_data text_size_small'>score</p>
                    <div className='table__hr'/>
                    <p className='text table__text_size-w_data text_size_small'>lines</p>
                    <div className='table__hr'/>
                    <p className='text table__text_size-w_data text_size_small'>level</p>
                    <div className='table__hr'/>
                    <p className='text table__text_size-w_data text_size_small'>state</p>
                </div>
                <hr className='box__hr-h'></hr>
                <ul className='ul ul_size-h_big'>
                    {data.map((player, index) => (
                        <li className='ul__li'>
                            <div className='location_horizontal-center'>
                                <p className='text table__text_size-w_index text_size_small'>{index + 1}</p>
                                <p className='text table__text_size-w_name text_size_small'>{player.name}</p>
                                <p className='text table__text_size-w_data text_size_small'>{player.score}</p>
                                <p className='text table__text_size-w_data text_size_small'>{player.lines}</p>
                                <p className='text table__text_size-w_data text_size_small'>{player.level}</p>
                                <p className='text table__text_size-w_data text_size_small'>{player.isPlay ? 'play' : 'came out'}</p>
                            </div>
                        </li>
                    ))}
                </ul>
            </div>
        </div>
    )
}

export default ResultsTable
