// components/statisticGame.js
import React, {useContext, useEffect, useState} from 'react'
import 'css/global.css'
import {levelRules} from 'services/config';

const StatisticsGame = ({statisticsRef, isPlay}) => {
    const [statistics, setStatistics] = useState({
        level: 1,
        score: 0,
        lines: 0,
        recordScore: 0,
        recordLines: 0,
        speed: levelRules[1].speed
      });

    useEffect(() => {
        statisticsRef.current = [statistics, setStatistics];
    })

    return (
        <div className='box box_size-w_small'>
            <ul className='ul ul_type_statistics'>
                {
                    isPlay
                    ?
                    <div>
                        <div className='line'>
                            <p className='line__text text text_size_small'>level: {statistics.level}</p>
                        </div>
                        <hr className='box__hr-h' />
                        <div className='line'>
                            <p className='line__text text text_size_small'>score: {statistics.score}</p>
                        </div>
                        <hr className='box__hr-h' />
                        <div className='line'>
                            <p className='line__text text text_size_small'>lines: {statistics.lines}</p>
                        </div>
                        <hr className='box__hr-h' />
                        <div className='line'>
                            <p className='line__text text text_size_small'>speed: {statistics.speed}</p>
                        </div>
                    </div>
                    :
                    <div/>
                }
                <hr className='box__hr-h' />
                <div className='line'>
                    <p className='line__text text text_size_small'>record score: {statistics.recordScore}</p>
                </div>
                <hr className='box__hr-h' />
                <div className='line'>
                    <p className='line__text text text_size_small'>record lines: {statistics.recordLines}</p>
                </div>
            </ul>
        </div>
    )
}

export default StatisticsGame
