// services/config.js
export const figures = [
    {
        name: 'Orange Ricky', 
        shape: [
            [0, 0, 1],
            [1, 1, 1]
        ]
    }, {
        name: 'Blue Ricky', 
        shape: [
            [1, 0, 0],
            [1, 1, 1]
        ]
    }, {
        name: 'Cleveland Z',
        shape: [
            [1, 1, 0],
            [0, 1, 1]
        ]
    }, {
        name: 'Rhode Lsland Z', 
        shape: [
            [0, 1, 1],
            [1, 1, 0]
        ]
    }, {
        name: 'Hero', 
        shape: [
            [1, 1, 1, 1]
        ]
    }, {
        name: 'Teewee', 
        shape: [
            [0, 1, 0],
            [1, 1, 1]
        ]
    }, {
        name: 'Smashboy', 
        shape: [
            [1, 1],
            [1, 1]
        ]
    }
]

export const colors = ['#76A639', '#A67439', '#6939A6', '#A6397A', '#39A69F', '#3971A6', '#A63959', '#A63939']

export const levelRules = {
    1: {
      scorePerLine: 10,
      speed: 500,
      nextlevelRulescore: 20,
    },
    2: {
      scorePerLine: 15,
      speed: 400,
      nextlevelRulescore: 500,
    },
    3: {
      scorePerLine: 20,
      speed: 300,
      nextlevelRulescore: 1000,
    },
    4: {
      scorePerLine: 30,
      speed: 200,
      nextlevelRulescore: 2000,
    },
    5: {
      scorePerLine: 50,
      speed: 100,
      nextlevelRulescore: Infinity,
    },
  }
