// pages/gameStart/components/roomPlayers.js
import React from 'react'
import ImageRoomGame from 'assets/roomGame.png'
import ImageDelete from 'assets/delete.png'
import ImageOk from 'assets/ok.png'
import ImageKingRoom from 'assets/kingRoom.png'

const RoomPlayers = () => {
    const propss = {isMainInRoom: true}
    propss.room = {
        name: 'room',
        players: [
            {
                id: 0,
                name: 'utoome1',
                isMainInRoom: true,
                isExcluted: false,
                isPlay: false
            }, {
                id: 1,
                name: 'utoome2',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: false
            }, {
                id: 2,
                name: 'utoome3',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: false
            }, {
                id: 3,
                name: 'utoome4',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: true
            }, {
                id: 4,
                name: 'utoome5',
                isMainInRoom: false,
                isExcluted: true,
                isPlay: true
            }, {
                id: 5,
                name: 'utoome6',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: false
            }, {
                id: 6,
                name: 'utoome7',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: false
            }, {
                id: 7,
                name: 'utoome8',
                isMainInRoom: false,
                isExcluted: false,
                isPlay: true
            }
            
        ]
    }

    return (
        <div className='box-room'>
            <div className='box-room__image-wrapper'>
                <img className='box-room__image tag_untouchable'
                    src={ImageRoomGame}/>
            </div>
            
            <div className='box-room__ul-wrapper line'>
                <ul className='box-room__ul line_ul'>
                    {propss.room.players.map((human, index) => (
                        <li className='ul__li'>
                            <div className='box-room_item'>
                                <p className='box-room_text'>{index + 1}) {human.name}</p>
                                {
                                    !(propss.isMainInRoom) && (human.isMainInRoom)
                                    ?
                                    <button className='button button_hover_to-react'>
                                        <img className='image_theme_small-icon' src={human.isMainInRoom ? ImageKingRoom : human.isExcluted ? ImageOk : ImageDelete}/>
                                    </button>
                                    :
                                    undefined
                                }
                                {
                                    (propss.isMainInRoom)
                                    ?
                                    <button className='button button_hover_to-react'>
                                        <img className='image_theme_small-icon' src={human.isMainInRoom ? ImageKingRoom : human.isExcluted ? ImageOk : ImageDelete}/>
                                    </button>
                                    :
                                    undefined
                                }
                            </div>
                        </li>
                    ))}
                </ul>
            </div>
            <div className='box-room__name-room-wrapper'>
                <p className='box-room__name-room tag_untouchable'>
                    {propss.room.name}
                </p>
            </div>
            
        </div>
    )
}

export default RoomPlayers
